drape_frontend/message_subclasses.hpp

```diff
@@ -405,28 +406,26 @@ class SetAddNewPlaceModeMessage : public Message
 {
 public:
   SetAddNewPlaceModeMessage(bool enable, std::vector<m2::TriangleD> && boundArea,
-                            bool enableKineticScroll, bool hasPosition, m2::PointD const & position)
+                            bool enableKineticScroll, m2::PointD const * optionalPosition)
     : m_enable(enable)
     , m_boundArea(std::move(boundArea))
     , m_enableKineticScroll(enableKineticScroll)
-    , m_hasPosition(hasPosition)
-    , m_position(position)
+  , m_position(optionalPosition ? *optionalPosition : decltype(m_position){})
```

Comment:
nit: I'd prefer in ctor:
```
if (optionalPosition)
  m_position = *optionalPosition;
```

Reply:
Ok. The initial idea was to make some variables const, the if doesn't allow that.